### PR TITLE
Subscription pass-through implementation and integration test

### PR DESCRIPTION
**Pull request by tomikazi**
Still WIP:
- more tests
- adding support for poll

### CODE REVIEW

File: pkg/northbound/gnmi/v2/subscribe.go [hunk 1]
```diff
@@ -1,14 +1,182 @@
...
+	"github.com/onosproject/onos-api/go/onos/topo"
+	"github.com/onosproject/onos-lib-go/pkg/errors"
+	baseClient "github.com/openconfig/gnmi/client"
 	"github.com/openconfig/gnmi/proto/gnmi"
+	"io"
+	"strings"
 )
 
+type subContext struct {
+	ctx       context.Context
```
Comment (adibrastegarnia) [inline pkg/northbound/gnmi/v2/subscribe.go:21]: we should not store context in struct. However it should be passed from one function to another and should flow through your program

FYI https://go.dev/blog/context-and-structs

File: pkg/southbound/gnmi/client.go [hunk 2]
```diff
@@ -31,15 +32,35 @@ type Client interface {
...
 }
 
-// Subscribe calls gNMI subscription based on a given query
+// Subscribe calls gNMI subscription bacc
+//sed on a given query
 func (c *client) Subscribe(ctx context.Context, q baseClient.Query) error {
 	err := c.client.Subscribe(ctx, q)
+	c.lock.Lock()
+	defer c.lock.Unlock()
+	if !c.running {
```
Comment (adibrastegarnia) [inline pkg/southbound/gnmi/client.go:46]: I am trying to understand the usage of this flag at client level. are you trying to achieve one Subscription stream per target?

**Review by adibrastegarnia — CHANGES_REQUESTED** [review]

File: pkg/northbound/gnmi/v2/subscribe.go [hunk 3]
```diff
@@ -1,14 +1,182 @@
...
+	nbStream  gnmi.GNMI_SubscribeServer
+	sbStreams map[string]gnmi.GNMI_SubscribeClient
```
Comment (adibrastegarnia) [inline pkg/northbound/gnmi/v2/subscribe.go:23]: How this is used?

File: pkg/southbound/gnmi/client.go (hunk 2 above)
Comment (tomikazi) [reply pkg/southbound/gnmi/client.go:46]: Yes.

File: pkg/northbound/gnmi/v2/subscribe.go (hunk 3 above)
Comment (tomikazi) [reply pkg/northbound/gnmi/v2/subscribe.go:22]: Ah, yes. This is a remnant from when I thought I was going to track the sb streams. I've removed this.

File: pkg/northbound/gnmi/v2/subscribe.go (hunk 1 above)
Comment (tomikazi) [reply pkg/northbound/gnmi/v2/subscribe.go]: Done.

**Review by adibrastegarnia — APPROVED** [review]

**Pull request merged**